### DISCUSSION TITLE
How to use linguist in javascript? for syntax highlighting

### DISCUSSION THREAD

**Discussion by Thecode764** [github-linguist/linguist#6979] [General]
Just read the title

**Comment by lildude** [comment]
You can't. Linguist is not responsible for syntax highlighting. It is used for language detection with the syntax highlighting performed by a separate internal application. Linguist is also written in Ruby.

**Comment by Thecode764** [comment]
giscus is a comment system written in preact and next.js powered by github discussions

but highlights syntax without ruby

![Screenshot_20240805_172128](https://github.com/user-attachments/assets/8531348f-d95b-4eec-bdaf-2278ef34a6aa)

**Comment by Thecode764** [comment]
giscus uses primer.style for design but primer.style doesn't have any component for highlighting

**Comment by Thecode764** [comment]
primer is github's design system

**Comment by lildude** [comment]
giscus isn't rendering the HTML for the syntax highlighting itself. It's using the raw HTML GitHub has already rendered. This is coming across in the [`bodyHTML`](https://github.com/giscus/giscus/blob/a40b386aa408f14c314ed78cac3c9551ae81a435/services/github/getDiscussion.ts#L72) GraphQL attribute it's requesting.

**Reply by Thecode764** [reply]
with github api?

**Reply by lildude** [reply]
Yup. They're [using](https://github.com/giscus/giscus/blob/a40b386aa408f14c314ed78cac3c9551ae81a435/services/github/getDiscussion.ts#L156) the [GraphQL API](https://docs.github.com/en/graphql)

**Reply by Thecode764** [reply]
thank you
